qiita_ware/test/test_commands.py

```diff
@@ -31,6 +34,65 @@
 
 
 @qiita_test_checker()
+class SSHTests(TestCase):
+    def setUp(self):
+        self.self_dir_path = path.dirname(path.abspath(__file__))
+        self.remote_dir_path = join(self.self_dir_path,
+                                    'test_data/test_remote_dir/')
+        self.test_ssh_key = join(self.self_dir_path, 'test_data/test_key')
+        self.test_wrong_key = join(self.self_dir_path, 'test_data/random_key')
+        self.temp_local_dir = mkdtemp()
+        self.exp_files = ['test_0.fastq.gz', 'test_1.txt']
+
+    def tearDown(self):
+        rmtree(self.temp_local_dir)
+
+    def _get_valid_files(self, folder):
+        files = []
+        for x in qiita_config.valid_upload_extension:
+            files.extend([basename(f) for f in glob(join(folder, '*.%s' % x))])
+        return files
+
+    def test_list_scp_wrong_key(self):
+        """Tests remote file listing using a wrong private key and scp"""
+        with self.assertRaises(AuthenticationException):
+            list_remote('scp://localhost:'+self.remote_dir_path,
+                        self.test_wrong_key)
+
+    def test_list_scp_nonexist_key(self):
+        """Tests remote file listing using a missing private key and scp"""
+        with self.assertRaises(IOError):
+            list_remote('scp://localhost:'+self.remote_dir_path,
+                        join(self.self_dir_path, 'nokey'))
+
+    def test_list_scp(self):
+        """Tests remote file listing using private key and scp"""
+        read_file_list = list_remote('scp://localhost:'+self.remote_dir_path,
+                                     self.test_ssh_key)
+        self.assertEqual(read_file_list, self.exp_files)
+
+    def test_list_sftp(self):
+        """Tests remote file listing using private key and sftp"""
+        read_file_list = list_remote('sftp://localhost:'+self.remote_dir_path,
+                                     self.test_ssh_key)
+        self.assertEqual(read_file_list, self.exp_files)
+
+    def test_download_scp(self):
+        """Tests remote file listing using private key and scp"""
+        download_remote('scp://localhost:'+self.remote_dir_path,
+                        self.test_ssh_key, self.temp_local_dir)
+        remote_filename_list = self._get_valid_files(self.remote_dir_path)
+        self.assertEqual(remote_filename_list, self.exp_files)
+
+    def test_download_sftp(self):
+        """Tests remote file listing using private key and sftp"""
```

Comment:
Although this was my previous comment:

> should be Tests remote file downloading ...

I think we stopped adding docstrings to test cases so I would suggest removing them altogether.

Reply:
K